fix bug handling zeek logs that have a _path column

mccanne opened the pull request.

aswan commented on pkg/zio/zeekio/parser.go:
```diff
@@ -212,37 +213,40 @@ func Unflatten(columns []zeek.Column, addPath bool) ([]zeek.Column, error) {
+		needpath = true
 	}
-	return cols, nil
+	return cols, needpath, nil
```
Nit: it looks like this never returns a non-nil error.  Since you're changing the signature here (and any dependent packages that call this function will need to be updated), lets remove the error from the signature at the same time.

mccanne replied:
done

mccanne commented:
> A test would be good!

test added

aswan commented on pkg/zio/ndjsonio/parser.go:
```diff
@@ -70,7 +70,7 @@ func (p *Parser) jsonParseObject(b []byte) (zeek.Type, error) {
+	columns, _ = zeekio.Unflatten(columns, false)
 	if err != nil {
 		return nil, err
 	}
```
now this can go

mccanne replied:
oops, done

aswan commented on tests/formats/autodetect/path.in.log:
```diff
@@ -0,0 +1,8 @@
+#path	conn
+#fields	_path	foo
+#types	string	string
+conn	yes
```
To really test that this is working right, we should use a different value than the `#path` directive here...

mccanne replied:
done

mccanne commented:
> nit: autodetect/ doesn't seem like the right place for the test...

I created a new subdir "path", moved the test there, and added a second test.
